lib/_emerge/actions.py

```diff
@@ -99,6 +99,10 @@
 from _emerge.UseFlagDisplay import pkg_use_display
 from _emerge.UserQuery import UserQuery
 
+emergerc_script = os.path.join("/", portage.const.USER_CONFIG_PATH, "emergerc")
+def callemergerc(phase):
+    os.system(" [[ -f " + emergerc_script + " ]] && export EMERGE_PHASE=" + phase + " && " + os.path.join(". /", emergerc_script))
```

Comment:
Don't use `os.system` for this, especially if `emergerc` is expected to be a bash script (not POSIX-compatible sh).

The `portage.process.spawn_bash` function might be more appropriate if `emergerc` is to be sourced as a bash script.

Alternatively, we could require that `emergerc` be executable and call `portage.process.spawn` or `subprocess.run` on it directly.

The `EMERGE_PHASE` environment var could be passed in via the `env` keyword argument.

Instead of testing for the existence of `emergerc_script` in bash, you could do it in python directly using `os.path.isfile`.

Reply:
Hi, thanks for you suggestions.

That's a lot of options, and I don't really understand the pros and cons for them. Which one would you prefer the most? 

I will do the isFile check in python, no problemo 👌

Reply:
I think I would opt for `subprocess.run`, since we don't really have a need for the more advanced features of `portage.process.spawn`.